templates/vpc.yml

```diff
@@ -58,6 +58,7 @@ Resources:
 
   Routes:
     Type: AWS::EC2::RouteTable
+    DependsOn: Gateway
```

Comment:
I have a feeling these need to be `DependsOn: GatewayAttachment` rather than `Gateway`.

The dependency on `Gateway` is implicit to the the `GatewayId: { Ref: Gateway }` below, whereas there's no way for CloudFormation to know about the dependency on `GatewayAttachment`.

The `AWS::EC2::VPCGatewayAttachment` is the thing that associates the `AWS::EC2::InternetGateway` with the `AWS::EC2::VPC`. Until the attachment is made, the Gateway is just “floating”, explaining the “[route table and internet gateway] belong to different networks” error message.

Reply:
*(PR inbound)*